### PR TITLE
add columnindex for DataFrameRow, speed up columnindex, fix a bug in indexing

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/DataFrames.jl/issues/2379

### CODE REVIEW

File: src/other/tables.jl [hunk 1]
```diff
@@ -6,6 +6,21 @@ Tables.rows(df::AbstractDataFrame) = eachrow(df)
...
+        loc = get(parent_ind.lookup, idx, 0)
+        return loc == 0 || loc > length(ind.remap) ? 0 : max(0, ind.remap[loc])
+    end
+end
+
+Tables.columnindex(df::Union{AbstractDataFrame, DataFrameRow}, idx::AbstractString) =
+    columnindex(df, Symbol(idx))
+
 Tables.columnindex(df::AbstractDataFrame, idx::AbstractString) =
     columnindex(df, Symbol(idx))
```
Comment (nalimilan) [inline src/other/tables.jl:25]: This method isn't needed anymore:
```suggestion
```

File: test/indexing.jl [hunk 2]
```diff
@@ -1822,6 +1822,19 @@ end
...
     end
+
+    @testset "cornercase of view indexing" begin
+        df = DataFrame(reshape(1:12, 4, :))
+        dfr = df[1, 3:2]
+        for idx in [:x1, :x2, :x3, :x4]
+            @test_throws ArgumentError dfr[idx]
+        end
+
+        sdf = df[1:1, 3:2]
```
Comment (nalimilan) [inline test/indexing.jl:1833]: Did you mean to do this?
```suggestion
        sdf = df[1:1, 3:2]
        dfr = view(df, 1, 3:2)
```

File: test/indexing.jl [hunk 3]
```diff
@@ -1822,6 +1822,19 @@ end
...
+        for idx in [:x1, :x2, :x3, :x4]
+            @test_throws ArgumentError dfr[idx]
```
Comment (bkamins) [inline test/indexing.jl:1835]:
```suggestion
            @test_throws ArgumentError sdf[idx]
```

**Comment by bkamins** [comment]
Thank you for the comments. Fixed

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**